src/vivarium/framework/engine.py

```diff
@@ -112,6 +113,15 @@ def __init__(
         model_specification = build_model_specification(
             model_specification, component_configuration, configuration, plugin_configuration
         )
+        results_dir = (
+            model_specification.to_dict()
+            .get("configuration", {})
+            .get("output_data", {})
+            .get("results_directory")
+        )
+        if results_dir:
+            with open(f"{results_dir}/complete_model_specification.yaml", "w") as f:
```

Comment:
Happy to take suggestions on what to call this. We already have a "model_specification.yaml" copied here as well (which is the model spec input we provide to `simulate run`).

Reply:
Why are we doing this in `__init__` rather than in `run_simulation`? This doesn't seem like the right place, since this means that every simulation (including the `InteractiveContext`) will write to disk, which doesn't seem right.

Regarding the name, I'd suggest comparing this to what we output in psimulate, which I believe prints out the entire model spec.

Reply:
hmm, good call

Reply:
Actually, I don't think it would - it only writes out to disk if a results dir is provided (ie via `simulate run` OR if a user went out of their way to add one to the config...in which case I think it should be saved out)